### PR TITLE
[WIN32SS][NTGDI] Refactoring PATH_WidenPath

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

This PR is a preparation to support pen width.
JIRA issue: [CORE-2527](https://jira.reactos.org/browse/CORE-2527), [CORE-13534](https://jira.reactos.org/browse/CORE-13534)

## Proposed changes
- Split IntGdiWidenPath function from PATH_WidenPath function.

```c
PPATH FASTCALL
IntGdiWidenPath(PPATH pPath, UINT penWidth, UINT penStyle, FLOAT eMiterLimit);

static PPATH FASTCALL PATH_WidenPath(DC *dc);
```
NOTE: IntGdiWidenPath is independent from DC.

### CODE REVIEW

File: win32ss/gdi/ntgdi/path.c [hunk 1]
```diff
@@ -1719,93 +1720,30 @@ PATH_StrokePath(
...
-    ExFreePoolWithTag(elp, TAG_PATH);
+    DWORD joint, endcap, penType;
 
     endcap = (PS_ENDCAP_MASK & penStyle);
     joint = (PS_JOIN_MASK & penStyle);
     penType = (PS_TYPE_MASK & penStyle);
 
     /* The function cannot apply to cosmetic pens */
-    if (obj_type == GDI_OBJECT_TYPE_EXTPEN && penType == PS_COSMETIC)
+    if (penType == PS_COSMETIC)
```
Comment (learn-more) [inline win32ss/gdi/ntgdi/path.c]: Before this checked also that the type was not {{GDI_OBJECT_TYPE_PEN}}.
This check was wrong?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/path.c]: I added a check in PATH_WidenPath. IntGdiWidenPath doesn't need this check. Thanks.

**Comment by learn-more** [comment]
https://reactos.org/testman/diff.php?id1=39132490&id2=39139609&type=1&strip=1

**Comment by katahiromz** [comment]
fixed
Fixed.

**Comment by katahiromz** [comment]
https://jira.reactos.org/secure/attachment/49274/WidenPathTest.zip
This is a test for WidenPath.

WIN10:
widenpathtest-win10
ROS (BEFORE):
widenpathtest-ros-before
ROS (AFTER):
widenpathtest-ros-after
There is no change between BEFORE and AFTER. PolyFillMode is wrong but not relevalant to this PR.

**Pull request merged**